src/main/java/com/google/devtools/build/lib/skyframe/ActionExecutionFunction.java

```diff
@@ -532,19 +534,43 @@ private SkyFunction.Reset handleLostInputs(
               env,
               actionStartTimeNanos);
     } catch (ActionRewindException rewindingFailedException) {
-      throw new ActionExecutionFunctionException(
-          new AlreadyReportedActionExecutionException(
-              skyframeActionExecutor.processAndGetExceptionToThrow(
-                  env.getListener(),
-                  e.getPrimaryOutputPath(),
-                  action,
-                  new ActionExecutionException(
-                      e,
-                      action,
-                      /* catastrophe= */ false,
-                      rewindingFailedException.getDetailedExitCode()),
-                  e.getFileOutErr(),
-                  ActionExecutedEvent.ErrorTiming.AFTER_EXECUTION)));
+      // If rewinding failed, Bazel may still be able to recover by retrying the invocation in
+      // BlazeCommandDispatcher if retries are enabled.
+      if (skyframeActionExecutor.invocationRetriesEnabled()) {
+        env.getListener().post(LostInputsEvent.create(e.getLostInputs().keySet()));
+        throw new ActionExecutionFunctionException(
```

Comment:
Can you share more code and limit the if/else to just differing behavior (posting the event, choosing the detailed exit code)?